### DISCUSSION TITLE
Is there any way to add code to the active theme files from a plugin?

### DISCUSSION THREAD

**Discussion by artxewebponferrada** [GetPublii/Publii#1409] [Help]
I'm working on some functionalities like the possibility to add custom fields in each post (for example custom css or js that comes out in only that post).

I manage to do it by editing directly the theme files but I would like to know if there is a way to do it from the main.js of a custom plugin. 

add something like this in the active theme

{{#checkIf @config.custom.frontSource '==' "post" }}}
{{> headhome}}
{{> top}}
   {{> frontpage-post}}
{{else}}
{{> head}}

**Comment by dziudek** [comment]
Hi,

You can consider `customHeadCode` insertion https://github.com/GetPublii/Publii/discussions/1359 for this purpose

**Comment by artxewebponferrada** [comment]
Thanks for the link, I have read it but I can not do one thing. Let's say that if I want to have some custom fields that appear in the configuration of each post (that is to say that we add them on the postConfig) would it be possible?

I want to "add" something like this (which is actually in the config.json file of the theme)

"postConfig": [
                {
            "name": "customHeadPost",
            "label": "Custom head post",
            "value": "",
            "type": "text",
            "group": "customHead post"
                },

**Reply by dziudek** [reply]
@artxewebponferrada - plugins API currently does not support such feature, but it is in our ToDo list, because we will also need this feature for some future planned plugins :)
